### ISSUE TITLE
Aria label for Make favourite action

### ISSUE THREAD

**Issue by edenhernandez-suse** [rancher/dashboard#8147]
## Context
There is an action to make a section a "favourite:

![image](https://user-images.githubusercontent.com/120640736/218424388-324e155a-964d-4e0f-b378-edb49efdcd3d.png)

This is what VoiceOver says:

![image](https://user-images.githubusercontent.com/120640736/218424475-15d33d7b-1023-4398-b1cb-7055e0acc914.png)

This information does not have enough context for users with visual impairments. 

## Expected Result
This component should have an aria-label with more context about what it does, e.g:
"Pin section to the main menu"

Part of #7968
